### DISCUSSION TITLE
FGTransmission Help Request

### DISCUSSION THREAD

**Discussion by jd60fe** [JSBSim-Team/jsbsim#413] [Q&A]
Greeting to the gurus!

I am trying to figure out how to use the FGTransmission class in my helicopter model to act as the gearbox between the engine(s) and the FGRotor I have defined.  I have searched the oracle (Google) and cannot find an example of its use.  The documentation doesn't show an "in-use" example and the User's Reference document is a bit dated now.

The actual use case is a UH-1N using two T400-CP-400 engines into the Combining Gearbox (CGB) which in turn drives the main gearbox to drive the main rotor and the tail rotor is driven by two smaller gearboxes that pull power off the main gearbox.

Thanks!

**Comment by bcoconni** [comment]
Hi @jd60fe and thanks for using JSBSim,

The  `FGTransmission` class is used internally in `FGRotor`:
https://github.com/JSBSim-Team/jsbsim/blob/375f5be0fdfeb210f4cf66f938e3e6fd426c2966/src/models/propulsion/FGRotor.cpp#L180-L198
Its parameters are set via the `<rotor>` thruster that is used in the example rotorcraft `ah1s` that is supplied with JSBSim. You can review the following folder and files for more information:
* `aircraft/ah1s`
* `engine/ah1s_rotor.xml`
* `engine/ah1s_tail_rotor.xml`

Let us know if you need some further support or if the content of the above files is unclear.

**Reply by jd60fe** [reply]
Bertrand,

Thank you so much!  Does the order engines and thrusters are listed in the Propulsion section of the aircraft.xml affect how JSBSim sees them?  Is this the "virtual drivetrain"?  I am trying to make my helicopter files better and still learning JSBSim.  I sue the reference guide a lot but the basic flow of how JSBSim ingest data is lost on me right now.

JD60FE

**Reply by bcoconni** [reply]
> Does the order engines and thrusters are listed in the Propulsion section of the aircraft.xml affect how JSBSim sees them?

Engines and thrusters are stored internally and executed by JSBSim in the order in which they were defined in the XML definition files. However, unless explicit dependencies are coded by the user between the engines, this should not be significant as JSBSim simply sums up the forces and moments generated by the propulsion elements and we all know that addition is commutative :smile:
